src/api/api.js

```diff
@@ -0,0 +1,48 @@
+const BASE_URL = process.env.REACT_APP_EASYAUTH_APP_URL + '/tenantbackend';
+
+const commonAPICall = async (
+    PATH,
+    METHOD = 'GET',
+    BODY = null,
+    headers = {
+      'accept': '*/*',
+      'Content-Type': 'application/json',
+    },
+) => {
+  const FULLPATH = BASE_URL + PATH;
+  const response = await fetch(FULLPATH, {
+    method: METHOD,
+    body: BODY,
+    headers: headers,
+  });
+
+  return response;
+};
+
+const getAPI = async (PATH, headers) => {
+  const response = await commonAPICall(PATH, 'GET', null, headers);
+  response.data = await response.json();
+  return response;
+};
+
+const postAPI = async (PATH, DATA) => {
+  const serializedData = JSON.stringify(DATA);
+  const response = await commonAPICall(PATH, 'POST', serializedData);
+  try {
+    response.data = await response.json();
+  } catch (err) {
+    console.log(err);
+  }
+  return response;
+};
+
+const getProfile = async (token) => {
+  const headers = {Authorization: `Bearer ${token}`};
+  const response = await getAPI('/api/profile', headers);
+  if (!response.ok) {
+    throw new Error(`Request failed with status ${response.status}`);
```

Comment:
How is this handled? To test you can open network tab in the inspection tool in chrome then click on the button "No throttling" and switch to "Offline" and then check what happens.

Reply:
when response is not okay, it will console log the error with status code and the component will not be shown ie. it will return null.

Reply:
Here is a throw. My question is who is catching it ultimately?